### PR TITLE
factor out capture stuff

### PR DESCRIPTION
**Pull request by cpacejo**
In preparation for the fast-path refactor, this factors out the packet capture code from the inbound and outbound processors into a single function in a new module `fastpath`.

I've made a couple modifications as part of this:
* added "packets filtered" counters – these are packets we could have captured but chose not to due to the filter
* added an explicit ZDP "link-layer" header structure (vs. just writing a raw u8 value)
* optimize the batch processing by exiting early when we're out of buffers/queue space
* switch to RwLock to guard the filter state, so the accessor methods can be non-async
* remove the unused `direction` argument from the capture messages

There is one new TODO I introduced; I wasn't able to figure out an efficient way to _use_ the batch functionality, so that is a minor regression.  I will try to revisit that after finishing the fastpath reorg.

### CODE REVIEW

File: adapter/ph/src/rpc_worker.rs [hunk 1]
```diff
@@ -382,15 +382,15 @@ async fn set_capture_program(asm: &Assembly<'_>, str_message: String) -> String
 
     let mut return_message = format!("Program: {program} set\n");
     match cbpf_rs::BpfProgram::validate(&insn_vec) {
-        Ok(final_program) => asm.flow_control.set_program(final_program).await,
+        Ok(final_program) => asm.flow_control.set_program(final_program),
```
Comment (owenrhysthomas) [inline adapter/ph/src/rpc_worker.rs:385]: With the removal of this `await`, I think this function no longer needs to be `async`.

File: adapter/ph/src/rpc_worker.rs [hunk 2]
```diff
@@ -382,15 +382,15 @@ async fn set_capture_program(asm: &Assembly<'_>, str_message: String) -> String
...
         _ => return_message = format!("Invalid program recieved, program not set\n"),
     }
 
     return_message
 }
 
 async fn delete_capture_program(asm: &Assembly<'_>) -> String {
-    asm.flow_control.delete_program().await;
+    asm.flow_control.delete_program();
```
Comment (owenrhysthomas) [inline adapter/ph/src/rpc_worker.rs:393]: No longer needs to be `async` either, I believe.

**Review by owenrhysthomas — APPROVED** [review]
Nice!

File: adapter/ph/src/rpc_worker.rs (hunk 1 above)
Comment (cpacejo) [reply adapter/ph/src/rpc_worker.rs:385]: Oh!  Good catches!

**Pull request merged**